MapboxNavigation/NavigationViewController.swift

```diff
@@ -148,16 +148,29 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
     public var shouldManageApplicationIdleTimer = true
     
     /**
-     Enables highlighting of the destination building when arriving.
+     Enum denoting the types of the destination building highlighting on arrival.
      */
-    public var highlightsDestinationBuildings: Bool = false
+    public enum DestinationBuildingHighlightType: Int {
```

Comment:
This name is a bit long. What if we call it something like `WaypointStyle` and name its cases `pin`, `building`, and `extrudedBuilding`? Would that capture enough nuance? I think it would also make it easier to support non-building treatments in the future, such as making a particular address label more prominent.

Reply:
Such naming looks good to me, it's generic enough to give the ability to extend it with other options other than buildings.